sdk/core/Azure.Core.Amqp/src/AnnotatedMessageMetadataDictionary.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Core.Amqp
+{
+    #nullable enable
+    internal class AnnotatedMessageMetadataDictionary : IDictionary<string, object?>
```

Comment:
I'm not sure that I understand the purpose of this;  why not project a dictionary interface over the  `AmqpAnnotatedMessage` directly and skip the intermediate collection?   If there wasn't a representation for a specific bit of data on the AMQP message, what would we do with it if added to this dictionary?   Ignore?   Make it an application property?